### ISSUE TITLE
Crash: AWSS3TransferUtility.m line 1074

### ISSUE THREAD

**Issue by AMORFATI-TRL** [aws-amplify/aws-sdk-ios#1994]
**Describe the bug**
I've had this crash for a while running different versions of the SDK but it's still happening in version **2.11.0**
I noticed it was reported as fixed in #1249 but it's still happening in a different line number. 

Fatal Exception: NSInvalidArgumentException
Cannot read file at file:///var/mobile/Containers/Data/Application/EEF5A5F0-359D-49EE-8548-928378FA82C9/Library/Application%20Support/MeerkatSessionManager/sessions/89ADDE21-8D5B-4761-881F-3278AACBD982.1571162648

**To Reproduce**
Can't easily reproduce on my test devices as it doesn't happen always 

**Which AWS service(s) are affected?**
S3

**Expected behavior**
Not crashing 

**Screenshots**
NA

**Environment(please complete the following information):**
 - SDK Version: 2.11.0
 - Dependency Manager: Cocoapods
 - Swift Version : 5.1

**Device Information (please complete the following information):**
 - Device: iPhone XR
 - iOS Version: iOS 13.1.2

**Additional context**
```
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x1a315dc30 __exceptionPreprocess
1  libobjc.A.dylib                0x1a2e780c8 objc_exception_throw
2  CFNetwork                      0x1a645ed20 (Missing)
3  CFNetwork                      0x1a645da7c (Missing)
4  CFNetwork                      0x1a645e3d0 (Missing)
5  CFNetwork                      0x1a645eae4 cfnTranslateCFError
6  libdispatch.dylib              0x1a2e0500c _dispatch_client_callout
7  libdispatch.dylib              0x1a2e11804 _dispatch_lane_barrier_sync_invoke_and_complete
8  CFNetwork                      0x1a63a2038 (Missing)
9  CFNetwork                      0x1a645e334 (Missing)
10 AWSS3                          0x10318710c __55-[AWSS3TransferUtility createUploadTask:startTransfer:]_block_invoke + 1074 (AWSS3TransferUtility.m:1074)
11 AWSCore                        0x1030567f8 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke + 331 (AWSTask.m:331)
12 AWSCore                        0x103025288 __30+[AWSExecutor defaultExecutor]_block_invoke_2 + 66 (AWSExecutor.m:66)
13 AWSCore                        0x1030257b0 -[AWSExecutor execute:] + 131 (AWSExecutor.m:131)
14 AWSCore                        0x1030563f8 -[AWSTask runContinuations] + 308 (AWSTask.m:308)
15 AWSCore                        0x103055f60 -[AWSTask trySetResult:] + 247 (AWSTask.m:247)
16 AWSCore                        0x1030573d0 -[AWSTaskCompletionSource setResult:] + 45 (AWSTaskCompletionSource.m:45)
17 AWSCore                        0x103056a04 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke_2 + 340 (AWSTask.m:340)
18 AWSCore                        0x1030568ac __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke + 348 (AWSTask.m:348)
19 AWSCore                        0x103025288 __30+[AWSExecutor defaultExecutor]_block_invoke_2 + 66 (AWSExecutor.m:66)
20 AWSCore                        0x1030257b0 -[AWSExecutor execute:] + 131 (AWSExecutor.m:131)
21 AWSCore                        0x1030563f8 -[AWSTask runContinuations] + 308 (AWSTask.m:308)
22 AWSCore                        0x103055f60 -[AWSTask trySetResult:] + 247 (AWSTask.m:247)
23 AWSCore                        0x1030573d0 -[AWSTaskCompletionSource setResult:] + 45 (AWSTaskCompletionSource.m:45)
24 AWSCore                        0x103056a04 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke_2 + 340 (AWSTask.m:340)
25 AWSCore                        0x1030567f8 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke + 331 (AWSTask.m:331)
26 AWSCore                        0x103025288 __30+[AWSExecutor defaultExecutor]_block_invoke_2 + 66 (AWSExecutor.m:66)
27 AWSCore                        0x1030257b0 -[AWSExecutor execute:] + 131 (AWSExecutor.m:131)
28 AWSCore                        0x1030563f8 -[AWSTask runContinuations] + 308 (AWSTask.m:308)
29 AWSCore                        0x103055f60 -[AWSTask trySetResult:] + 247 (AWSTask.m:247)
30 AWSCore                        0x1030573d0 -[AWSTaskCompletionSource setResult:] + 45 (AWSTaskCompletionSource.m:45)
31 AWSCore                        0x1030568c8 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke + 354 (AWSTask.m:354)
32 AWSCore                        0x103025288 __30+[AWSExecutor defaultExecutor]_block_invoke_2 + 66 (AWSExecutor.m:66)
33 AWSCore                        0x1030257b0 -[AWSExecutor execute:] + 131 (AWSExecutor.m:131)
34 AWSCore                        0x1030563f8 -[AWSTask runContinuations] + 308 (AWSTask.m:308)
35 AWSCore                        0x103055f60 -[AWSTask trySetResult:] + 247 (AWSTask.m:247)
36 AWSCore                        0x1030573d0 -[AWSTaskCompletionSource setResult:] + 45 (AWSTaskCompletionSource.m:45)
37 AWSCore                        0x103056a04 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke_2 + 340 (AWSTask.m:340)
38 AWSCore                        0x1030568ac __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke + 348 (AWSTask.m:348)
39 AWSCore                        0x103025288 __30+[AWSExecutor defaultExecutor]_block_invoke_2 + 66 (AWSExecutor.m:66)
40 AWSCore                        0x1030257b0 -[AWSExecutor execute:] + 131 (AWSExecutor.m:131)
41 AWSCore                        0x1030563f8 -[AWSTask runContinuations] + 308 (AWSTask.m:308)
42 AWSCore                        0x103055f60 -[AWSTask trySetResult:] + 247 (AWSTask.m:247)
43 AWSCore                        0x1030573d0 -[AWSTaskCompletionSource setResult:] + 45 (AWSTaskCompletionSource.m:45)
44 AWSCore                        0x103056a04 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke_2 + 340 (AWSTask.m:340)
45 AWSCore                        0x1030567f8 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke + 331 (AWSTask.m:331)
46 AWSCore                        0x103025288 __30+[AWSExecutor defaultExecutor]_block_invoke_2 + 66 (AWSExecutor.m:66)
47 AWSCore                        0x1030257b0 -[AWSExecutor execute:] + 131 (AWSExecutor.m:131)
48 AWSCore                        0x1030563f8 -[AWSTask runContinuations] + 308 (AWSTask.m:308)
49 AWSCore                        0x103055f60 -[AWSTask trySetResult:] + 247 (AWSTask.m:247)
50 AWSCore                        0x1030573d0 -[AWSTaskCompletionSource setResult:] + 45 (AWSTaskCompletionSource.m:45)
51 AWSCore                        0x103056a04 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke_2 + 340 (AWSTask.m:340)
52 AWSCore                        0x1030567f8 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke + 331 (AWSTask.m:331)
53 AWSCore                        0x103025288 __30+[AWSExecutor defaultExecutor]_block_invoke_2 + 66 (AWSExecutor.m:66)
54 AWSCore                        0x1030257b0 -[AWSExecutor execute:] + 131 (AWSExecutor.m:131)
55 AWSCore                        0x1030563f8 -[AWSTask runContinuations] + 308 (AWSTask.m:308)
56 AWSCore                        0x103055f60 -[AWSTask trySetResult:] + 247 (AWSTask.m:247)
57 AWSCore                        0x1030573d0 -[AWSTaskCompletionSource setResult:] + 45 (AWSTaskCompletionSource.m:45)
58 AWSCore                        0x103056a04 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke_2 + 340 (AWSTask.m:340)
59 AWSCore                        0x1030568ac __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke + 348 (AWSTask.m:348)
60 AWSCore                        0x103025288 __30+[AWSExecutor defaultExecutor]_block_invoke_2 + 66 (AWSExecutor.m:66)
61 AWSCore                        0x1030257b0 -[AWSExecutor execute:] + 131 (AWSExecutor.m:131)
62 AWSCore                        0x1030563f8 -[AWSTask runContinuations] + 308 (AWSTask.m:308)
63 AWSCore                        0x103055f60 -[AWSTask trySetResult:] + 247 (AWSTask.m:247)
64 AWSCore                        0x1030573d0 -[AWSTaskCompletionSource setResult:] + 45 (AWSTaskCompletionSource.m:45)
65 AWSCore                        0x103056a04 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke_2 + 340 (AWSTask.m:340)
66 AWSCore                        0x1030568ac __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke + 348 (AWSTask.m:348)
67 AWSCore                        0x103025288 __30+[AWSExecutor defaultExecutor]_block_invoke_2 + 66 (AWSExecutor.m:66)
68 AWSCore                        0x1030257b0 -[AWSExecutor execute:] + 131 (AWSExecutor.m:131)
69 AWSCore                        0x1030563f8 -[AWSTask runContinuations] + 308 (AWSTask.m:308)
70 AWSCore                        0x103055f60 -[AWSTask trySetResult:] + 247 (AWSTask.m:247)
71 AWSCore                        0x1030573d0 -[AWSTaskCompletionSource setResult:] + 45 (AWSTaskCompletionSource.m:45)
72 AWSCore                        0x103070160 __61-[AWSURLSessionManager URLSession:task:didCompleteWithError:]_block_invoke + 417 (AWSURLSessionManager.m:417)
73 AWSCore                        0x103056d9c __63-[AWSTask continueWithExecutor:successBlock:cancellationToken:]_block_invoke + 398 (AWSTask.m:398)
74 AWSCore                        0x1030567f8 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke + 331 (AWSTask.m:331)
75 AWSCore                        0x103025288 __30+[AWSExecutor defaultExecutor]_block_invoke_2 + 66 (AWSExecutor.m:66)
76 AWSCore                        0x1030257b0 -[AWSExecutor execute:] + 131 (AWSExecutor.m:131)
77 AWSCore                        0x103056660 -[AWSTask continueWithExecutor:block:cancellationToken:] + 368 (AWSTask.m:368)
78 AWSCore                        0x103056cfc -[AWSTask continueWithExecutor:successBlock:cancellationToken:] + 394 (AWSTask.m:394)
79 AWSCore                        0x103056e24 -[AWSTask continueWithSuccessBlock:] + 404 (AWSTask.m:404)
80 AWSCore                        0x10306f2e0 -[AWSURLSessionManager URLSession:task:didCompleteWithError:] + 260 (AWSURLSessionManager.m:260)
81 CFNetwork                      0x1a654d878 _CFNetworkHTTPConnectionCacheSetLimit
82 Foundation                     0x1a352dc60 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__
83 Foundation                     0x1a342f7e8 -[NSBlockOperation main]
84 Foundation                     0x1a352ffbc __NSOPERATION_IS_INVOKING_MAIN__
85 Foundation                     0x1a342f464 -[NSOperation start]
86 Foundation                     0x1a35309e8 __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__
87 Foundation                     0x1a35304a8 __NSOQSchedule_f
88 libdispatch.dylib              0x1a2e113ac _dispatch_block_async_invoke2
89 libdispatch.dylib              0x1a2e0500c _dispatch_client_callout
90 libdispatch.dylib              0x1a2e0778c _dispatch_continuation_pop
91 libdispatch.dylib              0x1a2e06eec _dispatch_async_redirect_invoke
92 libdispatch.dylib              0x1a2e13bd4 _dispatch_root_queue_drain
93 libdispatch.dylib              0x1a2e14384 _dispatch_worker_thread2
94 libsystem_pthread.dylib        0x1a2e6aa60 _pthread_wqthread
95 libsystem_pthread.dylib        0x1a2e70c78 start_wqthread
```

**Comment by royjit** [comment]
Thank you for reaching out. I have raised a PR for #2003 for the crash fix.

**Comment by royjit** [comment]
This has been released in [2.12.1](https://github.com/aws-amplify/aws-sdk-ios/releases/tag/2.12.1)
Please let us know if you have any issues.
